beacon-chain/rpc/eth/validator/validator.go

```diff
@@ -491,29 +519,49 @@ func (vs *Server) ProduceBlindedBlock(ctx context.Context, req *ethpbv1.ProduceB
 			}, nil
 		}
 	}
-
-	// After Bellatrix, return blinded block.
-	optimistic, err := vs.OptimisticModeFetcher.IsOptimistic(ctx)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not determine if the node is a optimistic node: %v", err)
-	}
-	if optimistic {
-		return nil, status.Errorf(codes.Unavailable, "The node is currently optimistic and cannot serve validators")
-	}
-	b, err := vs.V1Alpha1Server.GetBeaconBlock(ctx, v1alpha1req)
-	if err != nil {
-		return nil, status.Error(codes.Unavailable, "Could not get block from prysm API")
+	bellatrixBlock, ok := v1alpha1resp.Block.(*ethpbalpha.GenericBeaconBlock_BlindedBellatrix)
+	if ok {
+		// After Bellatrix, return blinded block.
+		optimistic, err := vs.OptimisticModeFetcher.IsOptimistic(ctx)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not determine if the node is a optimistic node: %v", err)
+		}
+		if optimistic {
+			return nil, status.Errorf(codes.Unavailable, "The node is currently optimistic and cannot serve validators")
+		}
```

Comment:
This same code is duplicated for Capella. We can move this check before we start processing bellatrix and do it only once.